### PR TITLE
chore: fix profile-sqlite for stackslib

### PR DESCRIPTION
**Pull request by bestmike007**
### Description

This PR is to fix building stackslib with `-F profile-sqlite`.

image

### Checklist
- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

File: stackslib/src/util_lib/db.rs [hunk 1]
```diff
@@ -716,9 +716,12 @@ pub fn sqlite_open<P: AsRef<Path>>(
     flags: OpenFlags,
     foreign_keys: bool,
 ) -> Result<Connection, sqlite_error> {
-    let db = Connection::open_with_flags(path, flags)?;
+    #[cfg(feature = "profile-sqlite")]
+    let mut db = Connection::open_with_flags(path, flags)?;
     #[cfg(feature = "profile-sqlite")]
     db.profile(Some(trace_profile));
+    #[cfg(not(feature = "profile-sqlite"))]
+    let db = Connection::open_with_flags(path, flags)?;
```
Comment (jcnelson) [inline stackslib/src/util_lib/db.rs:724]: Can you consolidate these?

```rust
#[cfg(feature = "profile-sqlite")]
fn inner_connection_open<P: AsRef<Path>>(path: P, flags: OpenFlags) -> Result<Connection, sqlite_error> {
   let mut db = Connection::open_with_flags(path, flags)?;
   db.profile(Some(trace_profile));
   Ok(db)
}

#[cfg(not(feature = "profile-sqlite"))]
fn inner_connection_open<P: AsRef<Path>>(path: P, flags: OpenFlags) -> Result<Connection, sqlite_error> {
   Connection::open_with_flags(path, flags)
}

pub fn sqlite_open<P: AsRef<Path>>(path: P, flags: OpenFlags, foreign_keys: bool) -> Result<Connection, sqlite_error> {
   let db = inner_connection_open(path, flags)?;
   /* etc. */
}
```

Comment (bestmike007) [reply stackslib/src/util_lib/db.rs]: Done.

**Review by jcnelson — APPROVED** [review]

**Review by jbencin — APPROVED** [review]
Looks good to me

**Comment by jcnelson** [comment]
Hi @bestmike007, can you re-sign the CLA?  Thanks!

**Comment by bestmike007** [comment]
@jcnelson I resigned a few days ago.

**Pull request merged**